### PR TITLE
fix(apparmor/host): streamline host profile generation with container  template generation

### PR DESCRIPTION
**Pull request by daemon1024**
deprecating special handling of host profiles

**Purpose of PR?**:

Fixes #

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/enforcer/appArmorEnforcer.go [hunk 1]
```diff
@@ -579,7 +579,24 @@ func (ae *AppArmorEnforcer) UpdateAppArmorHostProfile(secPolicies []tp.HostSecur
...
+		}
+		if err := kl.Clone(secPolicy.Spec.File, &hostPolicy.Spec.File); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec file to sec policy construct")
+		}
+		if err := kl.Clone(secPolicy.Spec.Network, &hostPolicy.Spec.Network); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec network to sec policy construct")
+		}
+		hostPolicies = append(hostPolicies, hostPolicy)
+	}
+
```
Comment (Aryan-sharma11) [inline KubeArmor/enforcer/appArmorEnforcer.go:607]: do we need to clone capabilities rule too?

File: KubeArmor/enforcer/appArmorEnforcer.go [hunk 2]
```diff
@@ -579,7 +581,30 @@ func (ae *AppArmorEnforcer) UpdateAppArmorHostProfile(secPolicies []tp.HostSecur
...
+			ae.Logger.Warnf("Error cloning host policy spec process to sec policy construct")
+		}
+		if err := kl.Clone(secPolicy.Spec.File, &hostPolicy.Spec.File); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec file to sec policy construct")
+		}
+		if err := kl.Clone(secPolicy.Spec.Network, &hostPolicy.Spec.Network); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec network to sec policy construct")
+		}
+		if err := kl.Clone(secPolicy.Spec.Capabilities, &hostPolicy.Spec.Capabilities); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec network to sec policy construct")
```
Comment (Aryan-sharma11) [inline KubeArmor/enforcer/appArmorEnforcer.go]:
```suggestion
			ae.Logger.Warnf("Error cloning host policy spec capabilities to sec policy construct")
```

File: KubeArmor/enforcer/appArmorEnforcer.go [hunk 3]
```diff
@@ -579,7 +581,30 @@ func (ae *AppArmorEnforcer) UpdateAppArmorHostProfile(secPolicies []tp.HostSecur
...
+		}
+		if err := kl.Clone(secPolicy.Spec.Syscalls, &hostPolicy.Spec.Syscalls); err != nil {
+			ae.Logger.Warnf("Error cloning host policy spec network to sec policy construct")
```
Comment (Aryan-sharma11) [inline KubeArmor/enforcer/appArmorEnforcer.go]:
```suggestion
			ae.Logger.Warnf("Error cloning host policy spec syscalls to sec policy construct")
```

**Review by Aryan-sharma11 — COMMENTED** [review]
Tested locally working fine, just some nits to address.

**Review by rksharma95 — DISMISSED** [review]

**Review by DelusionalOptimist — APPROVED** [review]

**Review by Aryan-sharma11 — APPROVED** [review]

**Pull request merged**